src/workerd/api/node/node.h

```diff
@@ -23,7 +24,51 @@ class CompatibilityFlags : public jsg::Object {
   CompatibilityFlags() = default;
   CompatibilityFlags(jsg::Lock&, const jsg::Url&) {}
 
+  // The compatibility flags method here is called at most once per isolate realm.
+  // After that, the flags are cached and always readonly.
+  jsg::JsObject getCompatibilityFlags(jsg::Lock& js, workerd::CompatibilityFlags::Reader flags) {
+    auto obj = js.objNoProto();
+    auto dynamic  = capnp::toDynamic(flags);
+    auto schema = dynamic.getSchema();
+
+    auto on = js.strIntern("on");
+    auto off = js.strIntern("off");
+    bool skipExperimental = !flags.getWorkerdExperimental();
+
+    for (auto field : schema.getFields()) {
+      // If this is an experimental flag, we expose it only if the experimental mode
+      // is enabled.
+      auto annotations = field.getProto().getAnnotations();
+      bool skip = false;
+      if (skipExperimental) {
+        for (auto annotation : annotations) {
+          if (annotation.getId() == EXPERIMENTAl_ANNOTATION_ID) {
+            skip = true;
+            break;
+          }
+        }
+      }
+      if (skip) continue;
+
+      bool value = dynamic.get(field).as<bool>();
```

Comment:
Since type is defined, does this work?

```suggestion
      bool value = dynamic.get(field).as();
```

Reply:
Probably but I prefer the more verbose option here.